### PR TITLE
Rationalize references for Microsoft.Net.Compilers.netcore

### LINKED ISSUE #12674: Microsoft.Net.Compilers.netcore needs to be updated for netstandard
The package `Microsoft.Net.Compilers.netcore` currently has support for `dotnet` and `netcoreapp1.0`. `dotnet` should be switched to `netstandard1.0` so that the latest compilers can be used more easily with CLI projects until the msbuild transition.

### PR DESCRIPTION
**Pull request by agocke**
This package now only supports the  TFM. This is because the purpose of the package
to layout a .NET Core executable, which requires the netcoreapp references, including CoreCLR itself.

There are also numerous references that are now provided by inheritence from the compiler itself, so
those references have been removed from the Microsoft.Net.Compilers.netcore nuspec.

Fixes #12674

### CODE REVIEW

File: src/Compilers/Server/PortableServer/project.json [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+    // We don't actually target netstandard1.6; this is to work around https://github.com/dotnet/roslyn/issues/12458
+    "netstandard1.6": {
+      "imports": "portable-net452"
     },
-    "runtimes": {
-        "win7-x64": { },
-        "ubuntu.14.04-x64": { },
-        "osx.10.10-x64": { }
+    "NETCoreApp1.0": {
+      "imports": "portable-net452"
```
Comment (jaredpar) [inline src/Compilers/Server/PortableServer/project.json]: Do we still need to do this?  Entire app top to bottom should now be Net Standard.

**Comment by agocke** [comment]
test this please

**Comment by agocke** [comment]
@jaredpar Was that a 👍?

**Pull request merged**